data/intro missions.txt

```diff
@@ -451,3 +451,139 @@ conversation "end of intro missions"
+		log `Paid a visit to James on <origin> and told him about the Hai.`
+	on complete
+		conversation
+			`James is impressed by the friendly aliens and humans living side by side on <planet>. While you are introducing him to one of the Hai, an old spacer recognizes James and offers to make further introductions around the community on <planet>. You make your goodbyes once again, and prepare to continue on your way.`
+		log `Paid a visit to James on <origin> to tell him about the Hai. Brought him to <planet> afterwards.`
+
+
+
+mission "Intro: Follow Up - After Main Plot at Allhome"
+	invisible
+	landing
+	source "Allhome"
+	to offer
+		has "Intro: Follow Up - Hai Before Main Plot: done"
+		has "main plot completed"
+	to fail
+		has "Intro: Follow Up - Hai After Main Plot: offered"
+	on offer
+		conversation
+			`It has been some time since you brought James to <origin>. Do you want to check in and see how he is doing?`
+			choice
+				`	(Yes.)`
+				`	(Not right now.)`
+					defer
+			`	You meet up with James at the spaceport, where he is sharing a drink with some other merchant captains. The group disperses to let the two of you catch up on old times, so you fill James in on some of the details of the recent war and the role you ended up playing in it. You talk a bit about the Pug and what you might do next, now that the war is behind you. Soon it is time for you to go and the two of you part ways again.`
+				decline
+	on decline
+		log `Met with James on <origin>, and filled him in on recent events with the war and the Pug.`
+
+
+
+mission "Intro: Follow Up - After Main Plot"
+	invisible
+	landing
+	source "Hestia"
+	to offer
+		or
+			has "Intro [3 Freighter]: done"
+			has "Intro [3 Transport]: done"
+			has "Intro [3 Interceptor]: done"
+		has "main plot completed"
+		not "First Contact: Hai: offered"
+	on offer
+		conversation
+			`As you're coming in for a landing, you remember that James left you an open invitation to look him up. Do you want to check in and see how he is doing?`
+			choice
+				`	(Yes.)`
+				`	(Not right now.)`
+					defer
+			`	James is happy to come out to the spaceport and meet with you. You spend some time talking about recent events, and fill him in on some of the details of the recent war and the role you ended up playing in it. You talk a bit about the Pug and what you might do next, now that the war is behind you.`
+				decline
+	on decline
+		log `Met with James on <origin>, and filled him in on recent events with the war and the Pug.`
+
+
+
+mission "Intro: Follow Up - Hai After Main Plot"
+	name "Bring James to <planet>"
+	description "Take James to <destination> to meet the Hai."
+	passengers 1
+	landing
+	source "Hestia"
+	destination "Allhome"
+	to offer
+		or
+			has "Intro [3 Freighter]: done"
+			has "Intro [3 Transport]: done"
+			has "Intro [3 Interceptor]: done"
+		has "main plot completed"
+		has "First Contact: Hai: offered"
+	to fail
+		has "Intro: Follow Up - Hai Before Main Plot: offered"
```

Comment:
It is for efficiency.  If it is in To Offer, it will be constantly checked.  If it is moved to a failed condition, the game will no longer waste cycles testing for it on every landing.  Missions that are no longer possible should be moved to "failed"

Reply:
>It is for efficiency. If it is in To Offer, it will be constantly checked.

`to offer` conditions aren't a matter of efficiency. Again, this is a matter of you trying to do things differently than they are currently done. Changing the status quo is something to be done in its own PR, not in a PR seeking to add a mission that should fit the current style.

Reply:
I think my willingness to make changes to bring things in line with the rest of the mission data is well evidenced by how quickly and eagerly I've done so.

But, in this case, you can see examples of 'to fail' all through the data in cases exactly like this, where it is best to stop the offering conditions from being tested on every landing.  it is absolutely an efficiency improvement to fail missions rather than to constantly check for their conditions.  The first one I see in FW Start is the "FW Hope Recon 1C" and there are a ton more.  It could also have a `not` condition on `to offer` but it uses `to fail` exactly like this mission does.

If you want to look at only more recent examples of the "current status quo", there are three examples in "wanderers middle.txt".

I don't think that is really what the concern is, though.  I see from the toxicity here that I have stepped into someones little pond, and that is not how I need to spend my time at all.

Reply:
>The first one I see in FW Start is the "FW Hope Recon 1C" and there are a ton more...
>If you want to look at only more recent examples of the "current status quo", there are three examples in "wanderers middle.txt".

For the first example you gave, there's a to fail there because `"FW Hope Recon 1B"` and `"1C"` are offered at the **exact same time**, but only one needs to be completed, which is why 1C fails when 1B is done. That's different from the situation here, where you're having `"Intro: Follow Up - Hai After Main Plot"` fail if `"Hai Before Main Plot"` is offered, but `"Hai Before Main Plot"` **can not be offered** while `"Hai After Main Plot"` is active, meaning that `"Hai Before Main Plot"` should be a `not` under `to offer`, not a `has` under `to fail`. It's a completely different scenario to "FW Hope Recon".

The three cases of `to fail` in wanderer middle are the same scenario as the Hope Recon scenario. I think you're fundamentally misunderstanding the uses of `to fail` in the game. `to fail` is only used when two missions can be active at the same time, but you're using `to fail` in a case where both missions can not be offered at the same time, in which case having `not` under `to offer` is what is used.

And please, don't confuse criticism with toxicity, because closing a PR in the face of helpful criticism that you label toxic doesn't help anyone.